src/Client/OfficeInterop/OfficeInterop.fs

```diff
@@ -1038,6 +1038,80 @@ let addOutputColumn (excelTable:Table) (arcTable:ArcTable) (newBB:CompositeColum
 
         loggingList
 
+let addBuildingBlock (excelTable:Table) (arcTable:ArcTable) (newBB:CompositeColumn) (headerRange:Excel.Range) (selectedRange:Excel.Range) =
+
+    let rowCount = arcTable.RowCount + 1
+
+    let buildingBlockCells = Spreadsheet.CompositeColumn.toStringCellColumns newBB
+
+    let targetIndex =
+        let excelIndex = rebaseIndexToTable selectedRange headerRange
+
+        let headers = arcTable.ToExcelValues().[0] |> Array.ofSeq |> Array.map (fun header -> header.ToString())
+
+        if excelIndex < (float) (headers.Length - 1) then 
+
+            //We want to start looking after the chosen column in order to ignore the current column because it could be a main column
+            let headers = headers.[(int) excelIndex + 1..]
+
+            let mainColumNames =
+                // Get all cases of the union
+                Microsoft.FSharp.Reflection.FSharpType.GetUnionCases(typeof<CompositeHeader>)
+                |> Array.map (fun case -> case.Name)
+
+            let indices =
+                mainColumNames
```

Comment:
Not 100% sure whats happening here, but maybe do something like Array.choose with index

```fsharp
let headers : string [] = [|"Name"; "Age"; "City"|]
let mainColumnNames : string [] = [|"Name"; "Age"|] 

[
  for i in 0 .. headers.Length - 1 do
    let header = headers.[i]
    if mainColumnNames |> Array.exists (fun cName -> cName.StartsWith header) then
      i
]
// val it: int list = [0; 1]
```

Reply:
fixed